### PR TITLE
[Driver] Support fprofile-sample-use= for CL

### PR DESCRIPTION
**Pull request by HaohaiWen**
Sampling PGO has already been supported on Windows. This patch adds
/fprofile-sample-use= /fprofile-sample-use: /fno-profile-sample-use and
supports -fprofile-sample-use= for CL.

### CODE REVIEW

File: clang/docs/UsersManual.rst [hunk 1]
```diff
@@ -2660,7 +2660,7 @@ usual build cycle when using sample profilers for optimization:
 
      > clang-cl /O2 -gdwarf -gline-tables-only ^
        /clang:-fdebug-info-for-profiling /clang:-funique-internal-linkage-names ^
-       /fprofile-sample-use=code.prof code.cc /Fe:code /fuse-ld=lld /link /debug:dwarf
```
Comment (tcreech-intel) [inline clang/docs/UsersManual.rst:2663]: I take it this wasn't working? (Neither `/fprofile-sample-use` nor `/fuse-ld`?)

**Review by tcreech-intel — APPROVED** [review]
Looks good to me.

File: clang/docs/UsersManual.rst (hunk 1 above)
Comment (HaohaiWen) [reply clang/docs/UsersManual.rst:2663]: Yes. I checked options.td.
-fprofile-sample-use was not a CLOption and there's no /fuse-ld.

File: clang/include/clang/Driver/Options.td [hunk 2]
```diff
@@ -8484,6 +8484,9 @@ def _SLASH_fp_strict : CLFlag<"fp:strict">, HelpText<"">, Alias<ffp_model_EQ>, A
 def _SLASH_fsanitize_EQ_address : CLFlag<"fsanitize=address">,
   HelpText<"Enable AddressSanitizer">,
   Alias<fsanitize_EQ>, AliasArgs<["address"]>;
+def : CLJoined<"fno-profile-sample-use">, Alias<fno_profile_sample_use>;
```
Comment (MaskRay) [inline clang/include/clang/Driver/Options.td:8487]: Do we really need these aliases?

Comment (HaohaiWen) [reply clang/include/clang/Driver/Options.td:8487]: For me, /fprofile-sample-use= is needed.
 I saw other CL _EQ flags has _COL alias.
In fact, I don't really need /fno-profile-sample-use.

Comment (HaohaiWen) [reply clang/include/clang/Driver/Options.td]: Removed /fno-profile-sample-use.

**Comment by HaohaiWen** [comment]
I'd like to merge it in if no objection.

**Comment by MaskRay** [comment]
Adding CLOption to -fprofile-sample-use= suffices.
We don't need these CLJoined aliases. They are for MSVC options that are ported to clang.
For clang-specific options, we don't want to add unneeded aliases.

**Comment by HaohaiWen** [comment]
> Adding CLOption to -fprofile-sample-use= suffices. We don't need these CLJoined aliases. They are for MSVC options that are ported to clang. For clang-specific options, we don't want to add unneeded aliases.

Got it. I checked CL flags with MSVC. You're right.
Let me revert those SLASH flags.

**Pull request merged**